src/groth16/aggregate/commit.rs

```diff
@@ -0,0 +1,152 @@
+/// TODO global commitment doc and usage
+use ff::{Field, PrimeField};
+use groupy::{CurveAffine, CurveProjective};
+use rayon::prelude::*;
+
+use crate::bls::Engine;
+use crate::groth16::aggregate::inner_product;
+
+/// Key is a generic commitment key that is instanciated with g and h as basis,
+/// and a and b as powers.
+#[derive(Clone, Debug)]
+pub struct Key<G: CurveAffine> {
+    /// Exponent is a
+    pub a: Vec<G>,
+    /// Exponent is b
+    pub b: Vec<G>,
+}
+/// VKey is a commitment key used by the "single" commitment on G1 values as
```

Comment:
I know you write a lot of go code, but please remove the "Name is a" from the docs, this is never done in rust docs (and one personal note one of the most annoying lints I have ever seen in go)

Reply:
but THANK YOU for writing so many comments!